exchanges/cryptodotcom/cryptodotcom_websocket.go

```diff
@@ -0,0 +1,604 @@
+	resp.Method = publicRespondHeartbeat
+	if authConnection {
+		return cr.Websocket.AuthConn.SendJSONMessage(resp)
+	}
+	return cr.Websocket.Conn.SendJSONMessage(resp)
+}
+
+// WsAuthConnect represents an authenticated connection to a websocket server
+func (cr *Cryptodotcom) WsAuthConnect(ctx context.Context, dialer *websocket.Dialer) error {
+	if !cr.Websocket.CanUseAuthenticatedEndpoints() {
+		return fmt.Errorf("%v AuthenticatedWebsocketAPISupport not enabled", cr.Name)
+	}
+	err := cr.Websocket.AuthConn.Dial(dialer, http.Header{})
+	if err != nil {
+		return fmt.Errorf("%v Websocket connection %v error. Error %v", cr.Name, cryptodotcomWebsocketUserAPI, err)
+	}
+	cr.Websocket.Wg.Add(1)
+	go cr.wsFunnelConnectionData(cr.Websocket.AuthConn, true)
+	return cr.AuthenticateWebsocketConnection()
+}
+
+// AuthenticateWebsocketConnection authenticates the websocekt connection.
+func (cr *Cryptodotcom) AuthenticateWebsocketConnection() error {
+	creds, err := cr.GetCredentials(context.Background())
+	if err != nil {
+		return err
+	}
+	timestamp := time.Now()
+	req := &WsRequestPayload{
+		ID:     cr.Websocket.AuthConn.GenerateMessageID(true),
+		Method: publicAuth,
+		Nonce:  timestamp.UnixMilli(),
+	}
+	var hmac, payload []byte
+	signaturePayload := publicAuth + strconv.FormatInt(req.ID, 10) + creds.Key + strconv.FormatInt(timestamp.UnixMilli(), 10)
+	hmac, err = crypto.GetHMAC(crypto.HashSHA256,
+		[]byte(signaturePayload),
+		[]byte(creds.Secret))
+	if err != nil {
+		return err
+	}
+	req.APIKey = creds.Key
+	req.Signature = crypto.HexEncodeToString(hmac)
+	payload, err = cr.Websocket.AuthConn.SendMessageReturnResponse(req.ID, req)
+	if err != nil {
+		return err
+	}
+	var resp *RespData
+	err = json.Unmarshal(payload, &resp)
+	if err != nil {
+		return err
+	} else if resp == nil {
+		return errors.New("no valid response from server")
+	}
+	if resp.Code != 0 {
+		mes := fmt.Sprintf("error code: %d Message: %s", resp.Code, resp.Message)
+		if resp.DetailCode != "0" && resp.DetailCode != "" {
+			mes = fmt.Sprintf("%s Detail: %s %s", mes, resp.DetailCode, resp.DetailMessage)
+		}
+		return errors.New(mes)
+	}
+	return nil
+}
+
+// Subscribe sends a websocket subscription to a channel message through the websocket connection handlers.
+func (cr *Cryptodotcom) Subscribe(subscriptions []stream.ChannelSubscription) error {
+	return cr.handleSubscriptions("subscribe", subscriptions)
+}
+
+// Unsubscribe sends a websocket unsubscription to a channel message through the websocket connection handlers.
+func (cr *Cryptodotcom) Unsubscribe(subscriptions []stream.ChannelSubscription) error {
+	return cr.handleSubscriptions("unsubscribe", subscriptions)
+}
+
+// GenerateDefaultSubscriptions Adds default subscriptions to websocket to be handled by ManageSubscriptions()
+func (cr *Cryptodotcom) GenerateDefaultSubscriptions() ([]stream.ChannelSubscription, error) {
+	subscriptions := []stream.ChannelSubscription{}
```

Comment:
```suggestion
	var subscriptions []stream.ChannelSubscription
```
The reason is because if there are any errors, we're not allocating memory space. Just declaring it as a var is an empty declaration. Its minor, but its a good practice to continue doing in all PRs